fili-sql/src/main/java/com/yahoo/bard/webservice/sql/SqlResultSetProcessor.java

```diff
@@ -37,13 +37,13 @@
  */
 public class SqlResultSetProcessor {
     private static final Logger LOG = LoggerFactory.getLogger(SqlResultSetProcessor.class);
-    private final DruidAggregationQuery<?> druidQuery;
-    private final ApiToFieldMapper apiToFieldMapper;
-    private BiMap<Integer, String> columnToColumnName;
-    private List<String[]> sqlResults;
-    private final ObjectMapper objectMapper;
-    private final int groupByDimensionsCount;
-    private final SqlTimeConverter sqlTimeConverter;
+    protected final DruidAggregationQuery<?> druidQuery;
+    protected final ApiToFieldMapper apiToFieldMapper;
+    protected BiMap<Integer, String> columnToColumnName;
+    protected List<String[]> sqlResults;
+    protected final ObjectMapper objectMapper;
+    protected final int groupByDimensionsCount;
+    protected final SqlTimeConverter sqlTimeConverter;
```

Comment:
I don't like making all these fields protected. This introduces a lot of direct coupling that could bite us hard, especially since this is presumably being done so that customers can extend this class (so there's going to be some direct coupling across codebases).

Ideally, we'd figure out *what* these fields need to be exposed to do, and expose methods to accomplish that, which customers can override as needed.

At worst, we should at least expose `protected` getters instead of the fields directly. At least that way if we ever need to change these, we have some hope of providing backwards compatibility.

Reply:
This applies to all the other places where we make previously private fields protected.

Reply:
So we can keep them protected but expose them with getters?

Reply:
I'd rather they be `private`, and exposed with `protected` getters.

Reply:
Fixed